packages/desktop-client/src/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { createContext, useContext, type ReactNode } from 'react';
+import { useSelector } from 'react-redux';
+
+import { type State } from 'loot-core/client/state-types';
+
+type AuthContextType = {
+  hasPermission: (permission?: string) => boolean;
```

Comment:
Changed to use the types from the permissions.

For the hasPermission receiving a permission null/undefined is because we can have buttons that may not have a permission set:

![image](https://github.com/user-attachments/assets/ac6f3be9-6e21-4165-a755-c92dc8b1e568)

This way its easier to read this condition:
![image](https://github.com/user-attachments/assets/84bea209-9d7c-4a33-a597-657da81d1bba)

Reply:
I could change the button logic to not call hasPermission if permission is null/undefined. but I find this more readiable.